src/useSpeech/index.ts

```diff
@@ -0,0 +1,117 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useIsMounted } from '../useIsMounted';
+
+type SpeechOptions = {
+	voice?: SpeechSynthesisVoice;
+	rate: number;
+	pitch: number;
+	volume: number;
+	lang?: string;
+};
+
+export type ISpeechOptions = Partial<SpeechOptions>;
+
+export type VoiceInfo = Pick<SpeechSynthesisVoice, 'lang' | 'name'>;
+
+type SpeechPlayStatus = 'init' | 'play' | 'pause' | 'end' | 'error';
+
+export type ISpeechState = SpeechOptions & {
+	isPlaying: boolean;
+	status: SpeechPlayStatus;
+	voiceInfo: VoiceInfo;
+	errorMessage?: string;
+};
+
+/**
+ * Hook for playing text using the SpeechSynthesis API.
+ *
+ * @param {string} text - Text to play.
+ * @param {ISpeechOptions} options - Options for playing the text.
+ * @param {SpeechSynthesisVoice} [options.voice] - Voice to use for playing the text.
+ * @param {number} [options.rate] - Rate at which to play the text.
+ * @param {number} [options.pitch] - Pitch at which to play the text.
+ * @param {number} [options.volume] - Volume at which to play the text.
+ * @param {string} [options.lang] - Language of the text.
+ * @returns {ISpeechState} State object representing the current state of speech synthesis.
+ */
+
+export const useSpeech = (text: string, options: ISpeechOptions): ISpeechState => {
+	const isMounted = useIsMounted();
```

Comment:
There is no need in such check - react 18  doest have warning about updating unmounted component's state anymore.

Reply:
Addressed [here](https://github.com/react-hookz/web/pull/1378/commits/4562ae7363ff866070aa49da8e3607c4805a86c1)